From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 13 May 2022 18:41:15 +0800
Subject: gdk/broadway: Split out sources listing

This way, we can reuse these lists in other non-autotools Makefile-based
build systems.

Origin: upstream, 3.24.35, commit:ea3b599d7fe1135ad9c9b32d11b73a6c73b3b848
---
 gdk/broadway/Makefile.am              | 51 ++------------------------------
 gdk/broadway/gdk-broadway-sources.inc | 55 +++++++++++++++++++++++++++++++++++
 2 files changed, 58 insertions(+), 48 deletions(-)
 create mode 100644 gdk/broadway/gdk-broadway-sources.inc

diff --git a/gdk/broadway/Makefile.am b/gdk/broadway/Makefile.am
index a43a577..81725de 100644
--- a/gdk/broadway/Makefile.am
+++ b/gdk/broadway/Makefile.am
@@ -1,5 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 include $(top_srcdir)/Makefile.decl
+include gdk-broadway-sources.inc
 NULL =
 
 libgdkincludedir = $(includedir)/gtk-3.0/gdk
@@ -23,15 +24,7 @@ noinst_LTLIBRARIES = libgdk-broadway.la
 
 bin_PROGRAMS = broadwayd
 
-libgdkinclude_HEADERS = 	\
-	gdkbroadway.h
-
-libgdkbroadwayinclude_HEADERS = 	\
-	gdkbroadwaydisplay.h		\
-	gdkbroadwaywindow.h		\
-	gdkbroadwaycursor.h		\
-	gdkbroadwaymonitor.h		\
-	gdkbroadwayvisual.h
+libgdkinclude_HEADERS = $(GDK_PUBLIC_H_SRCS_BROADWAY)
 
 EXTRA_DIST += toarray.pl
 
@@ -46,40 +39,12 @@ broadwayjs.h: broadway.js rawinflate.min.js
 EXTRA_DIST += broadway.js rawinflate.min.js
 
 # built headers that don't get installed
-broadway_built_private_headers =	\
-	clienthtml.h			\
-	broadwayjs.h
-
 broadway_built_sources =		\
 	${broadway_built_private_headers}
 
 libgdk_broadway_la_SOURCES =		\
 	${broadway_built_private_headers}\
-	gdkcursor-broadway.c	\
-	gdkdevice-broadway.h	\
-	gdkdevice-broadway.c	\
-	gdkdevicemanager-broadway.h \
-	gdkdevicemanager-broadway.c \
-	gdkdisplay-broadway.c	\
-	gdkdisplay-broadway.h	\
-	gdkdnd-broadway.c	\
-	gdkeventsource.c	\
-	gdkeventsource.h	\
-	gdkglobals-broadway.c	\
-	gdkkeys-broadway.c	\
-	gdkmonitor-broadway.c	\
-	gdkmonitor-broadway.h	\
-	gdkproperty-broadway.c	\
-	gdkscreen-broadway.c	\
-	gdkscreen-broadway.h	\
-	gdkselection-broadway.c	\
-	gdktestutils-broadway.c	\
-	gdkvisual-broadway.c	\
-	gdkwindow-broadway.c	\
-	gdkwindow-broadway.h	\
-	gdkprivate-broadway.h   \
-	gdkbroadway-server.h	\
-	gdkbroadway-server.c
+	$(GDK_BROADWAY_NON_GENERATED_SOURCES)
 
 if OS_UNIX
 libgdk_broadway_la_LIBADD = \
@@ -87,16 +52,6 @@ libgdk_broadway_la_LIBADD = \
 	$(NULL)
 endif
 
-broadwayd_SOURCES = \
-	broadway-protocol.h		\
-	broadwayd.c 			\
-	broadway-server.h		\
-	broadway-server.c		\
-	broadway-buffer.c		\
-	broadway-buffer.h		\
-	broadway-output.h		\
-	broadway-output.c
-
 if OS_WIN32
 broadwayd_LDADD = $(GDK_DEP_LIBS) -lws2_32
 else
diff --git a/gdk/broadway/gdk-broadway-sources.inc b/gdk/broadway/gdk-broadway-sources.inc
new file mode 100644
index 0000000..dd047f9
--- /dev/null
+++ b/gdk/broadway/gdk-broadway-sources.inc
@@ -0,0 +1,55 @@
+# Public header list for GdkBroadway
+
+GDK_PUBLIC_H_SRCS_BROADWAY =		\
+	gdkbroadway.h
+
+libgdkbroadwayinclude_HEADERS = 	\
+	gdkbroadwaydisplay.h		\
+	gdkbroadwaywindow.h		\
+	gdkbroadwaycursor.h		\
+	gdkbroadwaymonitor.h		\
+	gdkbroadwayvisual.h
+
+# built headers that don't get installed
+
+broadway_built_private_headers =	\
+	clienthtml.h			\
+	broadwayjs.h
+
+# Other non-generated sources
+GDK_BROADWAY_NON_GENERATED_SOURCES =	\
+	gdkcursor-broadway.c	\
+	gdkdevice-broadway.h	\
+	gdkdevice-broadway.c	\
+	gdkdevicemanager-broadway.h \
+	gdkdevicemanager-broadway.c \
+	gdkdisplay-broadway.c	\
+	gdkdisplay-broadway.h	\
+	gdkdnd-broadway.c	\
+	gdkeventsource.c	\
+	gdkeventsource.h	\
+	gdkglobals-broadway.c	\
+	gdkkeys-broadway.c	\
+	gdkmonitor-broadway.c	\
+	gdkmonitor-broadway.h	\
+	gdkproperty-broadway.c	\
+	gdkscreen-broadway.c	\
+	gdkscreen-broadway.h	\
+	gdkselection-broadway.c	\
+	gdktestutils-broadway.c	\
+	gdkvisual-broadway.c	\
+	gdkwindow-broadway.c	\
+	gdkwindow-broadway.h	\
+	gdkprivate-broadway.h   \
+	gdkbroadway-server.h	\
+	gdkbroadway-server.c
+
+broadwayd_SOURCES = \
+	broadway-protocol.h		\
+	broadwayd.c 			\
+	broadway-server.h		\
+	broadway-server.c		\
+	broadway-buffer.c		\
+	broadway-buffer.h		\
+	broadway-output.h		\
+	broadway-output.c
